src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Dynamo.Graph.Nodes;
+using Dynamo.Graph.Workspaces;
+using Dynamo.Logging;
+using Dynamo.Models;
+using Dynamo.PythonMigration.Properties;
+using Dynamo.UI.Prompts;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Extensions;
+using PythonNodeModels;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Media;
+
+namespace Dynamo.PythonMigration
+{
+    public class PythonMigrationViewExtension : IViewExtension
+    {
+        private const string EXTENSION_NAME = "Python Migration";
+        private const string EXTENSION_GUID = "1f8146d0-58b1-4b3c-82b7-34a3fab5ac5d";
+        private const string PYTHON_SCRIPT_NODE_TYPE = "PythonScriptNode";
+
+        private ViewLoadedParams LoadedParams { get; set; }
+        private DynamoViewModel DynamoViewModel { get; set; }
+        private NotificationMessage IronPythonNotification { get; set; }
+        private WorkspaceModel CurrentWorkspace { get; set; }
+
+        private Dictionary<Guid, NotificationMessage> NotificationTracker = new Dictionary<Guid, NotificationMessage>();
+        private Dictionary<Guid, IronPythonInfoDialog> DialogTracker = new Dictionary<Guid, IronPythonInfoDialog>();
+
+        /// <summary>
+        /// Extension GUID
+        /// </summary>
+        public string UniqueId { get { return EXTENSION_GUID; } }
+
+        /// <summary>
+        /// Extension Name
+        /// </summary>
+        public string Name { get { return EXTENSION_NAME; } }
+
+        public void Shutdown()
+        {
```

Comment:
theres a problem with the current implementation for viewExtensions - they don't have dispose called on them directly, but they do have Shutdown called - can you add a dispose() call here.